packages/noya-state/src/reducers/application.ts

```diff
@@ -1114,6 +1123,35 @@ export function reducer(
         );
       });
     }
+    case 'setSelectedSwatchGroup': {
+      const [, value] = action;
+      return produce(state, (state) => {
+        state.selectedGroupSwatch = value;
+      });
+    }
+    case 'groupSwatches': {
+      const [, id, value] = action;
+
+      const ids = typeof id === 'string' ? [id] : id;
+      const { basename, dirname } = delimitedPath;
+
+      return produce(state, (state) => {
+        const array = state.sketch.document.sharedSwatches?.objects ?? [];
+
+        array.forEach((object: Sketch.Swatch) => {
+          if (!ids.includes(object.do_objectID)) return;
+
+          const prevGroup = dirname(object.name);
+          const name = basename(object.name);
+
+          const group = value
```

Comment:
It looks like this code should join the different path segments only if they exist, but I find it a little hard to follow. Will this work?

```ts
const newName = [prevGroup, value, name].filter(x => !!x).join('/')
object.name = newName
```

Possibly a future util, `delimitedPath.join`